src/coap_client_libcoap.c

```diff
@@ -391,7 +391,6 @@ static uint32_t golioth_content_type_to_coap_type(enum golioth_content_type cont
         case GOLIOTH_CONTENT_TYPE_CBOR:
             return COAP_MEDIATYPE_APPLICATION_CBOR;
         default:
```

Comment:
We should also add an explicit case for `GOLIOTH_CONTENT_TYPE_OCTET_STREAM` here (it can fall through). That makes the code a little more maintainable so it's easier to see at a glance what the intention is and makes the code more searchable. In the future, when we have static analysis, it will also tell the analyzer that we intend for that case to use the default value.